source/clear-linux/guides/maintenance/kernel-development.rst

```diff
@@ -0,0 +1,462 @@
+
+While most packages in Clear Linux are typically packaged using 
+`autospec`_, the kernel is not. This means control files provided 
+by autospec are not available and changes must be made manually.
+
+
+#. Open the Linux kernel package RPM SPEC file in an editor.
+
+   .. code-block:: bash
+
+      $EDITOR linux.spec
+
+
+#. Modify the Version, Release, and Source0 URL entries at the top of the 
+   file to change the version of Linux kernel that will be compiled.
+
+   A list of current and available kernel release can be found on 
+   `kernel.org`_.
+
+   .. code-block:: bash
+      
+      Name:           linux
+      Version:        4.20.8
+      Release:        696
+      License:        GPL-2.0
+      Summary:        The Linux kernel
+      Url:            http://www.kernel.org/
+      Group:          kernel
+      Source0:        https://cdn.kernel.org/pub/linux/kernel/v4.x/linux-4.20.8.tar.xz
+      Source1:        config
+      Source2:        cmdline
+
+      %define ktarget  native
+
+   .. note::
+      - Consider changing the Name from *linux* in the RPM spec file to easily 
+        identify a modified kernel.
+
+      - Consider changing the ktarget from *native* in the RPM spec file to easily 
+        identify a modified kernel.
+
+
+#. Commit and save the changes to the file.
+
+
+Pull a copy of the Linux kernel source code
+===========================================
+
+You will need to obtain a local copy of the source code to make modifications 
+against.
+
+
+#. Run make sources to pull the kernel source code specified in the RPM 
+   SPEC file. In the example, it downloads the 
+   :file:`linux-4.20.8.tar.xz` file.
+
+   .. code-block:: bash
+
+      make sources
+
+
+#. Extract the kernel source code archive. This will create a working copy of 
+   the Linux source which you can modify.
+
+   .. code-block:: bash
+
+      tar -xvf linux-4.20.8.tar.xz
+
+
+#. Navigate to the extracted directory. In this example, it has been 
+   extracted into a :file:`linux-4.20.8` directory.
+
+   .. code-block:: bash
+
+      cd linux-4.20.8/
+
+
+Customize the Linux kernel source
+*********************************
+
+After the kernel sources have been obtained, customizations to the kernel 
+configuration or source code can be made for inclusion with the kernel 
+build.
+
+These customizations are option optional.
+
+
+Modify kernel configuration 
+===========================
+
+The kernel source has many configuration options available to pick 
+support for different hardware and software features.
+
+These configuration values must be provided in the :file:`.config` file at 
+compile time.
+
+You will need to make modifications to the :file:`.config` file, and include 
+it in the kernel package. 
+
+
+#. Make sure you have followed the steps to 
+   `Pull a copy of the Linux kernel source code`_ and are in the kernel 
+   source working directory.
+
+
+#. If you have an existing :file:`.config` file from an old kernel, copy it 
```

Comment:
Why does the user need to generate a new config and not use the current one from Clear Linux?

Reply:
I had it this way originally but changed it to not force readers to use the kernel fragmenting w/ CL tooling based on a comment from @bwarden. I think kernel fragment worflow (pulling base+mandatory from kernel-fagment) would be a good topic for a blog post and link this article from.

In the mean time do you think it's a good idea to do a `cp ../config .config` into the kernel source code directory so that they're using the config with packaging as the base config instead of `make defconfig`?

Reply:
> do you think it's a good idea to do a `cp ../config .config` into the

Yes.